### DISCUSSION TITLE
In mobile app - CANT select any BMGS values as it shows empty

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#39] [Installation] [answered]
**We are facing the issue with the following scenario:**
1. Created a new framework (eg. abcfw) and added categories, terms etc..
2. Published the framework
3. Made this framework as the default framework to the channel(eg. xyzchannel)
4. Created a user called "dummyuser1" and mapped this user to the channel "xyzchannel"

**The expectation is:**
When the user "dummyuser1" logged in, he will be mapped to "xyzchannel" and when he goes to PROFILE->EDIT , he should see the values(ie. BMGS) in the framework "abcfw" (because that is the defaulf framework
**It is working fine in Portal**
BUT in the app, when the user "dummyuser1" logged in, and edit his profile, it only shows the framework name, it does NOT shows any value values(ie. terms).
Basically the user CANT select any BMGS values as it shows empty!!
cc: @swayangjit @santoshgandham

![Screenshot from 2022-07-06 14-42-12](https://user-images.githubusercontent.com/107101132/177715214-0f1d912e-71e4-49de-a490-ef7fd3214b5a.png)
![Screenshot from 2022-07-06 14-44-10](https://user-images.githubusercontent.com/107101132/177715262-b046b8b8-f50f-4043-8c38-4e6619765d1d.png)

**Comment by swayangjit** [comment, marked as answer]
@nidhikachhawa24  Can you please post the curl for framework details for that framework where you are not getting any BMGS?

**Reply by nidhikachhawa24** [reply]
curl --location --request GET 'https://dhruva.shikshalokam.org/api/framework/v1/read/pplactsandboxfw2?cache=false&mode=edit'

cc: @swayangjit

**Reply by swayangjit** [reply]
@nidhikachhawa24 As we have an assumption in the mobile app that one framework can contain one board. So the name of the board should match the name of the framework.
`frameworkResponse.result.framework.name` should be equal to `frameworkResponse.result.framework.categories[0].terms[0].name`

**Reply by nidhikachhawa24** [reply]
As per the client requirement, we cannot keep the board and framework names same. They have different requirement. So is there any other way to achieve this?
@swayangjit

**Reply by swayangjit** [reply]
@nidhikachhawa24 The Framework name is not shown anywhere in the UI. Can you please inform us what the requirement is?

**Reply by swayangjit** [reply]
@nidhikachhawa24  Can you please let us know if your requirements are sorted or you need some help in this ?

**Reply by nidhikachhawa24** [reply]
@swayangjit 
We will try creating framework on the dummy org and let you know .
Meanwhile... Help us understand one thing -
We have created a framework with multiple board values(that is the user requirement)  and as suggested , made sure that framework name and one of the board name is same.  These are getting correctly displayed in the Portal->profile page.(after user logged in)
But, in the app(logged user->profile page) , we are seeing only one board value(which is same as the framework name).
We are not seeing the other values(terms) that we have created in the board.
So, as per your earlier comment, we need to create multiple frameworks for each of the board values to display in the app.
That probably takes a lot of time and duplicating the effort (also definitely confusing).
How can we make sure that both portal and app behave the same way with the same setting(in this case looking at the defaultFramework field)?
Please let us know, whether this can be achieved by any config change / code change. if it requires any code, should we start  different forum topic for the same or can we address that in this same thread?

**Reply by swayangjit** [reply]
@nidhikachhawa24 As of now in the board list will only show the frameworks associated with that org. It will require a larger discussion to change the current assumption.

**Reply by nidhikachhawa24** [reply]
Thank you @swayangjit

**Reply by kameshbhr** [reply]
@luckynvdu @sharathkashyap I think this is an important item to prioritize. There seems to be two assumptions, which have to be removed:
1. Framework name is same as name of the first term in the first category - Not sure why we have this assumption in the first place!
2. A framework cannot have more than on terms in the first category - I think the framework service doesn't assume this, but there are few places in UI where we are assuming this. We have to remove this assumption.
